### PR TITLE
feat: notification width and position

### LINKED ISSUE #216: Better support for tablets
Hi! 👋 
      
Firstly, thanks for your work on this project! 🙂

I was facing an issue where the notifications in tablets are the size of the screen, so they are too big and don't look good.

This is a quick solution I found to have a better UI for notifications on tablets

Here is the diff that solved my problem:

```diff
diff --git a/node_modules/react-native-notificated/src/core/config.ts b/node_modules/react-native-notificated/src/core/config.ts
index 954db1c..46f7105 100644
--- a/node_modules/react-native-notificated/src/core/config.ts
+++ b/node_modules/react-native-notificated/src/core/config.ts
@@ -7,6 +7,7 @@ const notificationSideMargin = 14
 const initialOffsetY = -300
 const targetOffsetY = true ? 50 : 10
 const maxNotificationWidth = 343
+const maxNotificationWidthTablet = 420
 
 export const Constants = {
   maxLongPressDragDistance,
@@ -15,4 +16,5 @@ export const Constants = {
   targetOffsetY,
   isAndroid,
   maxNotificationWidth,
+  maxNotificationWidthTablet
 }
diff --git a/node_modules/react-native-notificated/src/core/renderers/GestureHandler.tsx b/node_modules/react-native-notificated/src/core/renderers/GestureHandler.tsx
index 5c67815..20258d4 100644
--- a/node_modules/react-native-notificated/src/core/renderers/GestureHandler.tsx
+++ b/node_modules/react-native-notificated/src/core/renderers/GestureHandler.tsx
@@ -22,8 +22,9 @@ type Props = {
 
 export const GestureHandler = ({ children, state, animationAPI }: Props) => {
   const { width } = useWindowDimensions()
+
   const notificationWidth = state.isPortaitMode
-    ? width - Constants.notificationSideMargin * 2
+    ? Math.min(width, Constants.maxNotificationWidthTablet) - Constants.notificationSideMargin * 2
     : Constants.maxNotificationWidth
   return (
     <PanGestureHandler
```

This is how it looks with the current version of the library:
Simulator Screen Shot - iPad Pro (12 9-inch) (6th generation) - 2023-07-19 at 18 37 05

It can be fixed by setting maxWidth in the custom notification:
Simulator Screen Shot - iPad Pro (12 9-inch) (6th generation) - 2023-07-19 at 18 33 16

But the problem is that the PanResponder is still full width so you can't tap on the back button when the notification is shown:
Simulator Screen Shot - iPad Pro (12 9-inch) (6th generation) - 2023-07-19 at 18 33 54

And this is how it looks after the change, you can tap on the back button without any issues:
Simulator Screen Shot - iPad Pro (12 9-inch) (6th generation) - 2023-07-19 at 18 32 40

### PR DESCRIPTION
**Pull request by DominikDanielewicz**
Closes #216 && #218

### CODE REVIEW

File: src/core/renderers/GestureHandler.tsx [hunk 1]
```diff
@@ -28,14 +29,36 @@ export const GestureHandler = ({
...
+    }
+    return hasNotificationWidth ? notificationWidthFromState : Constants.maxNotificationWidth
+  }
+
+  let notificationWidth: number
+  if (state.isPortrait) {
+    notificationWidth = isWidthWithinBounds ? notificationWidthFromState : getDefaultWidth()
+  } else {
+    notificationWidth = getMaxWidth()
+  }
```
Comment (PdoubleU) [inline src/core/renderers/GestureHandler.tsx:52]: I think we can improve here a little bit: 

1. I would recommend moving these computations outside - it get's a little messy here, but if we create an utility function with 2-3 parameters, we can import it here, pass numbers, and assign the result to notificationWidth
2. I think this might be a little more readable:

```suggestion
  const fullWidth = width - Constants.notificationSideMargin * 2 // let's give it a better name, this is a full width of the screen with the margins

  const initialNotificationWidth = state?.notificationWidth || Constants.maxNotificationWidth; // let's name it initialNotificationWidht and assign to it user's value or default maxNotificationWidth - this way we simplify later logic

  const isWidthWithinBounds = notificationWidthFromState <= width

  const  notificationWidth = state.isPortrait && isWidthWithinBounds ? notificationWidthFromState : fullWidth // this is the shorthand from the if statement we had before (if (state.isPortrait) and then isWidthWithinBound are equal to state.IsPortrait && isWidthWithinBounds - any other situation we should return maxWidth
 
```

Just let me know what do you think about these little improvements, we can sync :)

By the way: good job!

File: src/core/utils/pickers.ts [hunk 2]
```diff
@@ -4,38 +4,62 @@ import type { DefaultKeys, DefaultStylesConfigs } from '../../defaultConfig/type
...
-      return topPosition
+      topOffset = topPosition
+      break
+    case 'top-left':
+      topOffset = topPosition
+      leftOffset = 10
+      break
+    case 'top-right':
+      topOffset = topPosition
+      rightOffset = 10
```
Comment (PdoubleU) [inline src/core/utils/pickers.ts:41]: I was wondering if we could use here Constants.notificationSideMargin, that will be more consistent. 
What I mean by that:
1. Full screen notification has side margins with 14px each, value from the Constants
2. Side screen notification will be with margin 10 from left, or right

In that case these 'margins' should be treated as left/right padding of the entire screen, and should be te same for each element.

What are your thoughts on it? 🤔

File: src/core/utils/pickers.ts [hunk 3]
```diff
@@ -4,38 +4,62 @@ import type { DefaultKeys, DefaultStylesConfigs } from '../../defaultConfig/type
...
+      topOffset = windowHeight - (notificationHeight ? notificationHeight + extraSpace : 150)
+      rightOffset = 10
+      break
     default:
-      return topPosition
+      topOffset = topPosition
+      break
   }
+
+  return { top: topOffset, left: leftOffset, right: rightOffset }
```
Comment (PdoubleU) [inline src/core/utils/pickers.ts:62]: Well done, I like this change! 🎉

**Review by PdoubleU — CHANGES_REQUESTED** [review]

**Review by PdoubleU — CHANGES_REQUESTED** [review]
We need to update documentation as well :)

File: example/package.json [hunk 4]
```diff
@@ -8,7 +8,8 @@
     "start": "react-native start",
     "test": "jest",
     "lint": "eslint . --ext .js,.jsx,.ts,.tsx",
-    "postinstall": "patch-package"
+    "postinstall": "patch-package",
+    "clear": "yarn --clearCache"
```
[the remark this answers is not in the document]
Comment (DominikDanielewicz) [reply example/package.json]: I removed the unnecessary script

File: example/package.json [hunk 5]
```diff
@@ -19,7 +20,7 @@
     "@reduxjs/toolkit": "^1.7.1",
     "react": "17.0.2",
     "react-native": "0.68.5",
-    "react-native-gesture-handler": "^2.9.0",
+    "react-native-gesture-handler": "2.9.0",
```
[the remark this answers is not in the document]
Comment (DominikDanielewicz) [reply example/package.json]: I restored ^ in this package version

File: example/src/screens/DefaultExamples.tsx [hunk 6]
```diff
@@ -10,10 +10,18 @@ import { RemoveButton } from '../components/basicExamples/RemoveButton'
...
-  notificationPosition: 'top',
+  notificationWidth: 300,
   defaultStylesSettings: {
+    globalConfig: {
+      titleSize: 30,
+    },
+    successConfig: {
+      notificationPosition: 'center',
+      titleSize: 20,
+    },
```
[the remark this answers is not in the document]
Comment (DominikDanielewicz) [reply example/src/screens/DefaultExamples.tsx]: I restored the original version of the example

File: example/src/screens/DefaultExamples.tsx [hunk 7]
```diff
@@ -31,8 +39,16 @@ export const DefaultExamples = () => {
...
+                description: 'This is where the toast text goes.',
                 title: 'Success',
+                style: {
+                  titleSize: 10,
+                },
+              },
+              config: {
+                duration: 2000,
+                notificationPosition: 'top-right',
+                notificationWidth: 200,
```
[the remark this answers is not in the document]
Comment (DominikDanielewicz) [reply example/src/screens/DefaultExamples.tsx]: I restored the original shape of the example and I will prepare a video and images.

File: src/core/renderers/GestureHandler.tsx (hunk 1 above)
Comment (DominikDanielewicz) [reply src/core/renderers/GestureHandler.tsx]: During testing, I noticed that notifications always display at full width when the device isn't in portrait mode. Since I removed the check for portrait mode, it's working correctly.

File: src/core/utils/pickers.ts (hunk 2 above)
Comment (DominikDanielewicz) [reply src/core/utils/pickers.ts]: Good point. I've adjusted the margins to take values from the Constants and all margins now are the same.

File: src/core/utils/pickers.ts (hunk 3 above)
Comment (DominikDanielewicz) [reply src/core/utils/pickers.ts:62]: Thanks!

File: docs/docs/intro/default-variants-config/width.md [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+
+Now, all notifications in the app will be displayed with a width of 400 pixels, except for the `error` notification mentioned in the previous example.<br />
+That `error` notification will have a width of 500 pixels because local configuration overrides the global setting.<br />
+Of course, if you prefer, you can set the width locally without adjusting the global setting.<br/>
+(You can read more about props overwriting in the [ORDER OF SETTINGS OVERWRITING](../comprehensive-configuration/order-of-settings-overwriting) section)
+
+<br/>
+<br/>
+
+## 🔦 Position config priority
```
Comment (PdoubleU) [inline docs/docs/intro/default-variants-config/width.md:105]: Width config priority?

File: example/yarn.lock [hunk 9]
```diff
@@ -6016,10 +6016,10 @@ react-native-codegen@^0.0.18:
     jscodeshift "^0.13.1"
     nullthrows "^1.1.1"
 
-react-native-gesture-handler@^2.9.0:
-  version "2.12.0"
-  resolved "https://registry.yarnpkg.com/react-native-gesture-handler/-/react-native-gesture-handler-2.12.0.tgz#59ca9d97e4c71f70b9c258f14a1a081f4c689976"
-  integrity sha512-rr+XwVzXAVpY8co25ukvyI38fKCxTQjz7WajeZktl8qUPdh1twnSExgpT47DqDi4n+m+OiJPAnHfZOkqqAQMOg==
+react-native-gesture-handler@2.9.0:
```
Comment (PdoubleU) [inline example/yarn.lock:6019]: You can revert these changes as well, yarn.lock still reflects changes in package.json :)

File: yarn.lock [hunk 10]
```diff
@@ -8503,10 +8503,10 @@ react-native-codegen@^0.0.18:
     jscodeshift "^0.13.1"
     nullthrows "^1.1.1"
 
-react-native-gesture-handler@2.9.0:
-  version "2.9.0"
-  resolved "https://registry.yarnpkg.com/react-native-gesture-handler/-/react-native-gesture-handler-2.9.0.tgz#2f63812e523c646f25b9ad660fc6f75948e51241"
-  integrity sha512-a0BcH3Qb1tgVqUutc6d3VuWQkI1AM3+fJx8dkxzZs9t06qA27QgURYFoklpabuWpsUTzuKRpxleykp25E8m7tg==
+react-native-gesture-handler@^2.9.0:
```
Comment (PdoubleU) [inline yarn.lock:8506]: I think this is the same left-over from changes in package.json. Could we undo these? 

yarn.lock reflects all changes in package.json

File: example/yarn.lock (hunk 9 above)
Comment (DominikDanielewicz) [reply example/yarn.lock]: I changed it to reflect the initial state

File: yarn.lock (hunk 10 above)
Comment (DominikDanielewicz) [reply yarn.lock]: I changed it to reflect the initial state

File: docs/docs/intro/default-variants-config/width.md (hunk 8 above)
Comment (DominikDanielewicz) [reply docs/docs/intro/default-variants-config/width.md]: Fixed

**Review by PdoubleU — APPROVED** [review]
